### PR TITLE
ENH: Support Selecting ITKv5 as a build option

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

File: Libs/vtkITK/CMakeLists.txt [hunk 1]
```diff
@@ -45,6 +49,7 @@ set(${PROJECT_NAME}_ITK_COMPONENTS
   ITKRegionGrowing
   ITKThresholding
   ITKVTK
+  ${ITKv5_DeprecatedLib} # Needed for ITKv5 for the SimpleFastMutexLock is needed as an alias to
```
Comment (jcfr) [inline Libs/vtkITK/CMakeLists.txt]: as alias to which class ?

Comment (hjmjohnson) [reply Libs/vtkITK/CMakeLists.txt]: Will fix.  std::mutex .

File: CMakeLists.txt.orig [hunk 2]
```diff
@@ -0,0 +1,1272 @@
+# ==== Define cmake build policies that affect compilation and linkage default behaviors
```
Comment (dzenanz) [inline CMakeLists.txt.orig]: File committed accidentally?

File: Libs/vtkITK/CMakeLists.txt [hunk 3]
```diff
@@ -45,6 +49,11 @@ set(${PROJECT_NAME}_ITK_COMPONENTS
   ITKRegionGrowing
   ITKThresholding
   ITKVTK
+  # Needed for ITKv5 so that SimpleFastMutexLock is an alias to std::mutex in ITKv5
+  # once ITKv4 is not supported, then we can simply replace SimpleFastMutexLock with std::mutex
+  # the use the the DeprecatedLib allows building without many preprocessor conditionals
```
Comment (dzenanz) [inline Libs/vtkITK/CMakeLists.txt]: `the use the the` -> `the use of the`?

File: SuperBuild/External_ITK.cmake [hunk 4]
```diff
@@ -79,6 +93,20 @@ if(NOT DEFINED ITK_DIR AND NOT Slicer_USE_SYSTEM_${proj})
   set(EP_SOURCE_DIR ${CMAKE_BINARY_DIR}/${proj})
   set(EP_BINARY_DIR ${CMAKE_BINARY_DIR}/${proj}-build)
 
+  if( ${Slicer_ITK_VERSION_MAJOR} GREATER_EQUAL 5)
+    list(APPEND EXTERNAL_PROJECT_OPTIONAL_CMAKE_CACHE_ARGS
+      -DITK_LEGACY_REMOVE:BOOL=OFF   #<-- Allow LEGACY ITKv4 features for now.
+      -DITK_LEGACY_SILENT:BOOL=ON    #<-- Silence for initial ITKv5 migrations.
+      -DModule_ITKDeprecated:BOOL=ON #<-- Needed for ITKv5 now. (ITK MultiThreader backwards compatibility.)
```
Comment (dzenanz) [inline SuperBuild/External_ITK.cmake]: `ITKDeprecated` is mostly for MutexLock -> comment needs an update

Comment (hjmjohnson) [reply SuperBuild/External_ITK.cmake]: @dzenanz #include <itkMultiThreader.h> was where I identified that this was needed, and that is why the comment is as it is.  

johnsonhj@Hanss-MBP$ find . -name "itkMultiThreader.h"
./Modules/Compatibility/Deprecated/include/itkMultiThreader.h

**Comment by phcerdan** [comment]
Superseeded in #1096

**Pull request closed without merging**